src/auth/auth.service.ts

```diff
@@ -1,9 +1,39 @@
-import { Injectable } from '@nestjs/common';
+import { Injectable, UnauthorizedException } from '@nestjs/common';
 import { UserService } from '../user/user.service';
 import bcrypt from 'bcrypt';
 import { JwtService } from '@nestjs/jwt';
+import { UserCredentialsDto } from '../user/dto/user.credentials.dto';
+import { UserSafeDto } from '../user/dto/user.from.db.dto';
 
 @Injectable()
 export class AuthService {
   constructor(private userService: UserService, private jwtService: JwtService) {}
+
+  async validateUser(userCredentialsDto: UserCredentialsDto): Promise<UserSafeDto | null> {
```

Comment:
Instead of that consider using Partial<> with Omit to get rid of sensitive data. 
https://www.typescriptlang.org/docs/handbook/utility-types.html#omittype-keys